src/plugins/condition/components/Criterion.vue

```diff
@@ -149,52 +147,82 @@ export default {
             return type;
         }
     },
+    watch: {
+        telemetry: {
+            handler(newTelemetry, oldTelemetry) {
+                this.checkTelemetry();
+            },
+            deep: true
+        }
+    },
     mounted() {
         this.updateMetadataOptions();
     },
     methods: {
         checkTelemetry() {
-            if(this.criterion.telemetry &&
-                !this.telemetry.find((telemetryObj) => this.openmct.objects.areIdsEqual(this.criterion.telemetry, telemetryObj.identifier))) {
-                //telemetry being used was removed. So reset this criterion.
-                this.criterion.telemetry = '';
-                this.criterion.metadata = '';
-                this.criterion.input = [];
-                this.criterion.operation = '';
-                this.persist();
+            if(this.criterion.telemetry) {
+                if (this.criterion.telemetry === 'any' && this.criterion.telemetry === 'all') {
+                    this.updateMetadataOptions();
+                } else {
+                    if (!this.telemetry.find((telemetryObj) => this.openmct.objects.areIdsEqual(this.criterion.telemetry, telemetryObj.identifier))) {
+                        //telemetry being used was removed. So reset this criterion.
+                        this.criterion.telemetry = '';
```

Comment:
Potentially we can refactor clearDependentFields to handle this, or create a new resetValues method to handle both scenarios.